<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SearchWidget</class>
 <widget class="QFrame" name="SearchWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>812</width>
    <height>145</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="styleSheet">
    <string notr="true">
      QLabel { font: bold; }
    </string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>3</number>
   </property>
   <property name="bottomMargin">
    <number>3</number>
   </property>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="spacing">
        <number>9</number>
       </property>
       <property name="bottomMargin">
        <number>2</number>
       </property>
       <item>
        <widget class="QLabel" name="graphLabel">
         <property name="toolTip">
          <string>The graph on which the search will be performed</string>
         </property>
         <property name="text">
          <string>graph:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="scopeLabel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>The element type on which the search will be performed</string>
         </property>
         <property name="text">
          <string>scope:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="searchTermsLabel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>search for:</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <property name="spacing">
        <number>9</number>
       </property>
       <property name="leftMargin">
        <number>3</number>
       </property>
       <item>
        <widget class="TreeViewComboBox" name="graphCombo">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Choose the graph on which the search will be performed</string>
         </property>
         <property name="sizeAdjustPolicy">
          <enum>QComboBox::AdjustToContents</enum>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="spacing">
          <number>9</number>
         </property>
         <property name="topMargin">
          <number>2</number>
         </property>
         <item>
          <widget class="QComboBox" name="scopeCombo">
           <property name="maximumSize">
            <size>
             <width>350</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Choose the element type on which the search will be performed</string>
           </property>
           <property name="sizeAdjustPolicy">
            <enum>QComboBox::AdjustToContents</enum>
           </property>
           <item>
            <property name="text">
             <string>Nodes &amp; Edges</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Nodes only</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Edges only</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>Choose 'Do not modify selection' to only count the number of nodes/edges found</string>
           </property>
           <property name="text">
            <string>selection mode:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="selectionModeCombo">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>350</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Choose 'Do not modify selection' to only count the number of nodes/edges found</string>
           </property>
           <property name="sizeAdjustPolicy">
            <enum>QComboBox::AdjustToContents</enum>
           </property>
           <item>
            <property name="text">
             <string>Replace current selection</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Add to current selection</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Remove from current selection</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Do not modify selection</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="resultsStorageLabel">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>The graph property whose values associated with the found elements will be modified</string>
           </property>
           <property name="text">
            <string>store results in:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="resultsStorageCombo">
           <property name="maximumSize">
            <size>
             <width>350</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Choose the graph property whose values associated with the found elements will be modified</string>
           </property>
           <property name="sizeAdjustPolicy">
            <enum>QComboBox::AdjustToContents</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <property name="spacing">
          <number>9</number>
         </property>
         <item>
          <widget class="QComboBox" name="searchTermACombo">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>350</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Choose the property for which the search will performed</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="operatorCombo">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>350</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Choose the search comparison operator</string>
           </property>
           <item>
            <property name="text">
             <string>=</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>â‰ </string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>&gt;</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>â‰¥</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>&lt;</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>â‰¤</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>starting with</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>ending with</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>containing</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>matching (reg. exp.)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string> like (sql like filter)</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="searchTermBCombo">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>350</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Choose the value involved in the search comparison</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTableWidget" name="tableWidget">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>30</height>
            </size>
           </property>
           <property name="styleSheet">
            <string notr="true">QTableView {
border: 1px solid #C9C9C9;
color: black;
font: 12px;
}</string>
           </property>
           <property name="frameShape">
            <enum>QFrame::NoFrame</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Plain</enum>
           </property>
           <property name="lineWidth">
            <number>0</number>
           </property>
           <property name="midLineWidth">
            <number>0</number>
           </property>
           <property name="showGrid">
            <bool>false</bool>
           </property>
           <property name="cornerButtonEnabled">
            <bool>false</bool>
           </property>
           <attribute name="horizontalHeaderVisible">
            <bool>false</bool>
           </attribute>
           <attribute name="horizontalHeaderStretchLastSection">
            <bool>true</bool>
           </attribute>
           <attribute name="verticalHeaderVisible">
            <bool>false</bool>
           </attribute>
           <row>
            <property name="text">
             <string>New Row</string>
            </property>
           </row>
           <column>
            <property name="text">
             <string>New Column</string>
            </property>
           </column>
           <item row="0" column="0">
            <property name="text">
             <string/>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="spacing">
      <number>9</number>
     </property>
     <item>
      <widget class="QPushButton" name="searchButton">
       <property name="minimumSize">
        <size>
         <width>100</width>
         <height>0</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Execute the search</string>
       </property>
       <property name="statusTip">
        <string>Execute the search</string>
       </property>
       <property name="text">
        <string>Search</string>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/tulip/graphperspective/icons/22/edit-find.png</normaloff>:/tulip/graphperspective/icons/22/edit-find.png</iconset>
       </property>
       <property name="shortcut">
        <string>Return</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="caseSensitivityCheck">
       <property name="toolTip">
        <string>Indicate if strings comparison will be case sensitive</string>
       </property>
       <property name="statusTip">
        <string>Indicate if strings comparison will be case sensitive</string>
       </property>
       <property name="text">
        <string>case sensitive</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLabel" name="resultsCountLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="styleSheet">
        <string notr="true">font: bold 10px;</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TreeViewComboBox</class>
   <extends>QComboBox</extends>
   <header>tulip/TreeViewComboBox.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>searchTermACombo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>SearchWidget</receiver>
   <slot>termAChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>154</y>
    </hint>
    <hint type="destinationlabel">
     <x>494</x>
     <y>271</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>searchTermBCombo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>SearchWidget</receiver>
   <slot>termBChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>374</x>
     <y>154</y>
    </hint>
    <hint type="destinationlabel">
     <x>494</x>
     <y>271</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>searchButton</sender>
   <signal>clicked()</signal>
   <receiver>SearchWidget</receiver>
   <slot>search()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel">
     <x>494</x>
     <y>271</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>graphIndexChanged(int)</slot>
  <slot>termAChanged()</slot>
  <slot>termBChanged()</slot>
  <slot>search()</slot>
 </slots>
</ui>
